test/test_derive.py

```diff
@@ -0,0 +1,238 @@
+        ]
+        baseKeyPvtTemplate = [
+            (PyKCS11.CKA_CLASS, PyKCS11.CKO_PRIVATE_KEY),
+            (PyKCS11.CKA_KEY_TYPE, PyKCS11.CKK_ECDSA),
+            (PyKCS11.CKA_TOKEN, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_SENSITIVE, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_PRIVATE, PyKCS11.CK_FALSE),
+            (PyKCS11.CKA_DECRYPT, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_SIGN, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_UNWRAP, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_LABEL, "TestBaseKeyP256"),
+            (PyKCS11.CKA_ID, keyID),
+            (PyKCS11.CKA_DERIVE, PyKCS11.CK_TRUE),
+        ]
+        mechanism = PyKCS11.Mechanism(PyKCS11.CKM_EC_KEY_PAIR_GEN, None)
+        self.baseEcPubKey, self.baseEcPvtKey = self.session.generateKeyPair(baseKeyPubTemplate, baseKeyPvtTemplate, mechanism)
+        self.assertIsNotNone(self.baseEcPubKey)
+        self.assertIsNotNone(self.baseEcPvtKey)
+
+        # Known base symmetric key
+        knownAESKeyValue = bytes([0xAA, 0xAA, 0xAA, 0xAA, 0xBB, 0xBB, 0xBB, 0xBB,
+                                        0xCC, 0xCC, 0xCC, 0xCC, 0xDD, 0xDD, 0xDD, 0xDD])
+
+        keyID = (0x11,)
+        wrapKeyTemplate = [
+            (PyKCS11.CKA_CLASS, PyKCS11.CKO_SECRET_KEY),
+            (PyKCS11.CKA_KEY_TYPE, PyKCS11.CKK_AES),
+            (PyKCS11.CKA_TOKEN, PyKCS11.CK_FALSE),
+            (PyKCS11.CKA_SENSITIVE, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_PRIVATE, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_ENCRYPT, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_UNWRAP, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_VALUE_LEN, 16),
+            (PyKCS11.CKA_LABEL, "wrap"),
+            (PyKCS11.CKA_ID, keyID),
+        ]
+        mechanism = PyKCS11.Mechanism(PyKCS11.CKM_AES_KEY_GEN, None)
+        AESKey = self.session.generateKey(wrapKeyTemplate, mechanism)
+        self.assertIsNotNone(AESKey)
+
+        mechanism = PyKCS11.Mechanism(PyKCS11.CKM_AES_ECB, None)
+        wrappedKey = self.session.encrypt(AESKey, knownAESKeyValue, mechanism)
+        self.assertIsNotNone(wrappedKey)
+        keyID = (0x02,)
+        baseAESKeyTemplate = [
+            (PyKCS11.CKA_CLASS, PyKCS11.CKO_SECRET_KEY),
+            (PyKCS11.CKA_KEY_TYPE, PyKCS11.CKK_AES),
+            (PyKCS11.CKA_TOKEN, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_SENSITIVE, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_PRIVATE, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_ENCRYPT, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_DECRYPT, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_SIGN, PyKCS11.CK_FALSE),
+            (PyKCS11.CKA_VERIFY, PyKCS11.CK_FALSE),
+            (PyKCS11.CKA_DERIVE, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_EXTRACTABLE, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_VALUE_LEN, 16),
+            (PyKCS11.CKA_LABEL, "TestBaseAESKey"),
+            (PyKCS11.CKA_ID, keyID),
+        ]
+        self.baseAESKey = self.session.unwrapKey(AESKey, wrappedKey, baseAESKeyTemplate, mechanism)
```

Comment:
Maybe you should also check that `self.baseAESKey` is equal to `knownAESKeyValue` to check the wrap/unwrap worked?

Reply:
Good idea! Fixing.